skyportal/handlers/api/tag.py

```diff
@@ -0,0 +1,238 @@
+import re
+
+from sqlalchemy import func
+
+from baselayer.app.access import auth_or_token
+
+from ...models import Obj, ObjTag, ObjTagOption
+from ..base import BaseHandler
+
+
+class ObjTagOptionHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        with self.Session() as session:
+            tags = session.scalars(ObjTagOption.select(session.user_or_token)).all()
+            return self.success(data=tags)
+
+    @auth_or_token
+    def post(self):
+        data = self.get_json()
+        name = data.get("name")
+
+        if not name or not isinstance(name, str):
+            return self.error("`name` must be provided as a non-empty string")
+
+        if not re.fullmatch(r"[A-Za-z0-9]+", name):
+            return self.error(
+                "`name` must contain only letters and numbers (no spaces, underscores, or special characters)",
+                status=400,
+            )
+
+        with self.Session() as session:
+            existing_tag = session.scalars(
+                ObjTagOption.select(session.user_or_token).where(
+                    func.lower(ObjTagOption.name) == name.lower()
+                )
+            ).first()
+
+            if existing_tag:
+                return self.error(
```

Comment:
maybe also return the existing tag's name so the user can understand why it says it exists (as in it's identical to another one and providing it is a bit clearer than only saying case insensitive match).